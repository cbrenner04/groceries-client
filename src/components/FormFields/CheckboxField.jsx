import React from 'react';
import PropTypes from 'prop-types';

const CheckboxField = ({
  name,
  label,
  value,
  handleChange,
  classes,
}) => (
  <div className={`form-check ${classes}`}>
    <input
      className="form-check-input"
      id={name}
      type="checkbox"
      checked={value}
      onChange={handleChange}
    />
    <label className="form-check-label" htmlFor={name}>
      {label}
    </label>
  </div>
);

CheckboxField.propTypes = {
  name: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  value: PropTypes.bool,
  handleChange: PropTypes.func.isRequired,
  classes: PropTypes.string,
};

CheckboxField.defaultProps = {
  classes: '',
  value: false,
};

export default CheckboxField;
