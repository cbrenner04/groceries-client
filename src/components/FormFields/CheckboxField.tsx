import React, { type ChangeEventHandler } from 'react';
import { Form } from 'react-bootstrap';
import type { FormCheckType } from 'react-bootstrap/esm/FormCheck';

export interface ICheckboxFieldProps {
  name: string;
  label: string;
  value?: boolean;
  handleChange: ChangeEventHandler;
  classes?: string;
  type?: FormCheckType;
}

const CheckboxField: React.FC<ICheckboxFieldProps> = ({
  name,
  label,
  value = false,
  handleChange,
  classes = '',
  type = 'checkbox',
}): React.JSX.Element => (
  <Form.Group controlId={name} className={`${classes} mb-3`}>
    <Form.Check type={type} checked={value} onChange={handleChange} label={label} name={name} />
  </Form.Group>
);

export default CheckboxField;
