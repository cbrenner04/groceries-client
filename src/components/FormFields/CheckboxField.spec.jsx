import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import CheckboxField from './CheckboxField';

const defaultProps = {
  handleChange: jest.fn(),
  name: 'testName',
  label: 'testLabel',
  value: true,
};

describe('CheckboxField', () => {
  it('renders input', () => {
    const { getByLabelText } = render(<CheckboxField {...defaultProps} />);
    const formInput = getByLabelText(defaultProps.label);
    const formGroup = formInput.parentElement;

    expect(formGroup).toMatchSnapshot();
    expect(formInput).toBeChecked();
  });

  describe('when value changes', () => {
    it('calls handleChange', () => {
      const { getByLabelText } = render(<CheckboxField {...defaultProps} />);
      const formInput = getByLabelText(defaultProps.label);
      fireEvent.click(formInput);

      expect(defaultProps.handleChange).toHaveBeenCalled();
    });
  });
});
