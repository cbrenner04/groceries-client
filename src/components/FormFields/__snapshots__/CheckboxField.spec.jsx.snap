// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxField renders input 1`] = `
<div
  class="form-check"
>
  <input
    checked=""
    class="form-check-input"
    id="testName"
    type="checkbox"
  />
  <label
    class="form-check-label"
    for="testName"
    title=""
  >
    testLabel
  </label>
</div>
`;
